lib/code_corps/messages/messages.ex

```diff
@@ -67,9 +67,16 @@ defmodule CodeCorps.Messages do
   def create(%{} = params) do
     %Message{}
     |> Message.changeset(params)
+    |> Changeset.cast(params, [:author_id, :project_id])
+    |> Changeset.validate_required([:author_id, :project_id])
+    |> Changeset.assoc_constraint(:author)
+    |> Changeset.assoc_constraint(:project)
+    |> Changeset.cast_assoc(:conversations, with: &Messages.Conversations.create_changeset/2)
```

Comment:
`create` requires more work than the default `changeset` does. We should probably add more tests for the `create` function to make sure these changes work. I don't think there's an explicit need to extract a create changeset, though.

Reply:
There's an argument to be made here that this belongs in its own `changeset`, akin to `Messages.Conversations.create_changeset`.

Reply:
@begedin this is the nested POST, right?  I remember doing this once and I had to create a changeset.

I think what I did was take the nested ids, pass it through `|> Enum.map(&Changeset.change/1)` and then `put_assoc(changeset, :conversations, conversation_changesets)` or something like that.

Reply:
But that required the nested ids to already exist in the DB.  Not sure what the case is here.

Reply:
@snewcomer The ids are not in the database in this case. The child record is created alongside the parent.